src/common/nodes/TypeState.ts

```diff
@@ -105,9 +111,11 @@ export class TypeState {
             try {
                 instance = FunctionInstance.fromPartialInputs(
                     definition,
-                    (id): NonNeverType | undefined => {
+                    (
+                        id
+                    ): { output: NonNeverType | undefined; sequence: NonNeverType | undefined } => {
                         const edgeSource = getSourceType(n.id, id);
-                        if (edgeSource) {
+                        if (edgeSource.output && edgeSource.sequence) {
```

Comment:
I'm not convinced that this is correct. Please explain why you made this the new condition, specifically, why you check both.